types/handler.ts

```diff
@@ -0,0 +1,51 @@
+import { JsonRpcProvider } from "@ethersproject/providers";
+import { ChainId, networkCurrencies, networkExplorers, networkIds, networkNames, networkRpcs, tokens } from "./constants";
+
+export type ValidBlockData = {
+  jsonrpc: string;
+  id: number;
+  result: {
+    number: string;
+    timestamp: string;
+    hash: string;
+  };
+};
+
+export type Token = {
+  decimals: number;
+  address: string;
+};
+
+export type NativeToken = {
+  symbol: string;
+  decimals: number;
+};
+
+export type HandlerInterface = {
+  getProvider(): JsonRpcProvider | undefined;
```

Comment:
| undefined seems like a bad idea. At that point I would just use the optional property name syntax. But really I would use null most likely.